src/Symfony/Contracts/HttpClient/ResponseInterface.php

```diff
@@ -59,6 +59,18 @@ public function getHeaders(bool $throw = true): array;
      */
     public function getContent(bool $throw = true): string;
 
+    /**
+     * Gets the response body decoded as array, typically from a JSON payload.
```

Comment:
does this mean component will deal with e.g. XML if requested? What if ppl want to leverage a serializer for this?

Honestly, do want the API? :D

From #30502 

> ppl can always encode/decode on their own if they really want precise control on these

yet, we kinda want this 30+ line method :) should it be util instead?

Reply:
Integration with Serializer should be provided by decoration.

Reply:
Note I see no reasons why integration with Serializer should implement `HttpClientInterface`: turning string (even when coming from response bodies) to objects is not the domain of an http client to me.

Reply:
In that case i think we need to update the description to something like `Gets a decoded body array representation (typically ...).`

Let's say i decorate this getArray to mutate data, but im passing this client globally around in my app; some services might be effected by this.

Reply:
:thinking: I'm not sure to spot the difference, semantically :)

Reply:
Missed your comment, you're right we can expect HTTP domain; thus the body decoded as-is.

:+1: with a simple decoration solution for extension in the future.

Reply:
that would also help #30494 btw